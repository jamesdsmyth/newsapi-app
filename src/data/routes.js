// routes - the categories available from newsapi.org
const routes = [
  {
    route: 'home',
    category: 'general'
  },
  {
    route: 'business',
    category: 'business'
  },
  {
    route: 'entertainment',
    category: 'entertainment'
  },
  {
    route: 'health',
    category: 'health'
  },
  {
    route: 'science',
    category: 'science'
  },
  {
    route: 'sports',
    category: 'sports'
  },
  {
    route: 'technology',
    category: 'technology'
  }
]

export default routes;