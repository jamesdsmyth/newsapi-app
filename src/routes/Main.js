import React, { Component } from 'react';

import { Route, Redirect } from "react-router-dom";

import NewsLanding from '../containers/NewsLanding';
import NewsArticleLanding from '../containers/NewsArticleLanding';

// Main is the container that houses the routes of the app
class Main extends Component {
  render() {
    return (
      <main className="main">
        <Route exact path="/" render={() => (
            <Redirect to="/home"/>
          )}
        />
        {
          this.props.routes.map((item, index) => (
            <Route
              exact
              path={`/${item.route}`}
              key={index}
              render={() => (
                <NewsLanding category={item.category} />
              )}
            />
          ))
        }
        {
          this.props.routes.map((item, index) => (
            <Route
              path={`/${item.route}/:articleId`}
              key={index}
              render={(props) => (
                <NewsArticleLanding {...props} category={item.category} />
              )}
            />
          ))
        }
      </main>
    )
  }
}

export default Main;