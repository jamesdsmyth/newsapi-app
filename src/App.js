import React from 'react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from 'react-redux';

import store from './reducers/combinedReducers';

import routes from './data/routes';

import Header from './components/Header';
import Main from './routes/Main';

import './styles/app.scss';

const App = () => (
  <Provider store={store}>
    <Router>
      <div className="page-container">
        <Header routes={routes} />
        <Main routes={routes} />  
      </div>
    </Router>
  </Provider>
)

export default App;