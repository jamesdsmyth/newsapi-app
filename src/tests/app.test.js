import React from 'react';
import { mount, configure } from 'enzyme';

import ReactDOM from 'react-dom';
import expect from 'expect';
import store from '../reducers/combinedReducers'; 
import App from '../App';
import Adapter from 'enzyme-adapter-react-16';

import routes from '../data/routes';

configure({ adapter: new Adapter() });

describe('<App />', () => {

  const wrapper = mount(<App routes={routes} />);

  it('should load initially', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App store={store} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should have the correct wrapping class', () => {
    expect(wrapper.find('.page-container')).toBeDefined();
  });

  it('should contain a <Provider /> component', () => {
    expect(wrapper.find('Provider')).toBeDefined();
  });

  it('should contain a <Router /> component', () => {
    expect(wrapper.find('Router')).toBeDefined();
  });

  it('should contain a <Header /> component', () => {
    expect(wrapper.find('Header')).toBeDefined();
  });

  it('should contain a <Main /> component', () => {
    expect(wrapper.find('Main')).toBeDefined();
  });
});














// describe('<NewsLanding />', () => {
//   it('should render correctly when given basic data', () => {


//     const category = 'general';

//     const wrapper = shallow(
//       <NewsLanding
//         category={category}
//       />
//     );
//     expect(wrapper.find(NewsLanding)).to.have.lengthOf(1);
//     expect(wrapper.find('h1').length).toEqual(1);
//   });
// });

// describe('<NewsLanding />', () => {
    

//   it('should render an input, label and SVGIcon', () => {
//     const category = 'general';
//     const wrapper = shallow(
//       <NewsLanding
//         category={category}
//       />
//     );

//     expect(wrapper.find('h1').text()).toEqual(category);
//     // expect(wrapper.find('label').length).toEqual(1);
//     // expect(wrapper.find('SVGIcon').length).toEqual(1);
//     // expect(wrapper.find('input').length).toEqual(1);
//   });



  // it('should render correctly when given basic data', () => {
  //   // const badgeType = colors.CLASSES_TEACHERS;
  //   const value = 45;
  //   const category = 'general';

  //   const wrapper = shallow(
  //     <NewsLanding
  //       category={category}
  //     />
  //   );

  //   // check if correct style is added to the badge
  //   expect(wrapper.find(`.main-content`).length).toEqual(1);
  //   // span will contain the correct value
  //   // expect(wrapper.find('span').contains(value)).toEqual(true);
  // });
// });





























// describe('loading for elements on the page', () => {
//   it('app should load initially', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(<App store={store} />, div);
//     ReactDOM.unmountComponentAtNode(div);
//   });
// });

// describe('newsReducer actions', () => {
//   it('should return the initial state', () => {
//     expect(newsReducer(undefined, {})).toEqual({ status: 'loading' });
//   });

//   it('should handle UPDATE_NEWS_SUCCESS', () => {
//     const getNewsSuccessAction = {
//       type: 'UPDATE_NEWS_SUCCESS',
//       data: mockApiResponseSuccess
//     }

//     expect(newsReducer({}, getNewsSuccessAction)).toEqual({ articles: getNewsSuccessAction.data, status: 'ok' });
//   });

//   it('should handle UPDATE_NEWS_FAILURE', () => {
//     const getNewsFailureAction = {
//       type: 'UPDATE_NEWS_FAILURE',
//       data: mockApiResponseFailure
//     }

//     expect(newsReducer({}, getNewsFailureAction)).toEqual({ data: getNewsFailureAction.data, status: 'error' });
//   });
// });






  // it('should handle GET_POST_SUCCESS');
  // it('should handle UPDATE_POST_SUCCESS');
  // it('should handle GET_POST_FAIL');
  // it('should handle GET_POST_START');
