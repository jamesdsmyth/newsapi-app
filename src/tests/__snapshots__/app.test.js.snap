// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders <App /> and matches the snapshot 1`] = `
<div
  className="page-container"
>
  <header
    className="header-content header-content--hidden"
  >
    <section
      className="header-content__mobile-buttons"
    >
      <span
        className="header-content__mobile-buttons__menu"
        onClick={[Function]}
      >
        Menu
      </span>
      <span
        className="header-content__mobile-buttons__menu__hide"
        onClick={[Function]}
      >
        x
      </span>
    </section>
    <nav
      className="navigation"
    >
      <ul
        className="navigation__main"
      >
        <li
          className="navigation__main__list-item"
          onClick={[Function]}
        >
          <a
            aria-current="page"
            className="navigation__main__list-item__link navigation__main__list-item__link--active"
            href="/home"
            onClick={[Function]}
            style={Object {}}
          >
            home
          </a>
        </li>
        <li
          className="navigation__main__list-item"
          onClick={[Function]}
        >
          <a
            aria-current={null}
            className="navigation__main__list-item__link"
            href="/business"
            onClick={[Function]}
          >
            business
          </a>
        </li>
        <li
          className="navigation__main__list-item"
          onClick={[Function]}
        >
          <a
            aria-current={null}
            className="navigation__main__list-item__link"
            href="/entertainment"
            onClick={[Function]}
          >
            entertainment
          </a>
        </li>
        <li
          className="navigation__main__list-item"
          onClick={[Function]}
        >
          <a
            aria-current={null}
            className="navigation__main__list-item__link"
            href="/health"
            onClick={[Function]}
          >
            health
          </a>
        </li>
        <li
          className="navigation__main__list-item"
          onClick={[Function]}
        >
          <a
            aria-current={null}
            className="navigation__main__list-item__link"
            href="/science"
            onClick={[Function]}
          >
            science
          </a>
        </li>
        <li
          className="navigation__main__list-item"
          onClick={[Function]}
        >
          <a
            aria-current={null}
            className="navigation__main__list-item__link"
            href="/sports"
            onClick={[Function]}
          >
            sports
          </a>
        </li>
        <li
          className="navigation__main__list-item"
          onClick={[Function]}
        >
          <a
            aria-current={null}
            className="navigation__main__list-item__link"
            href="/technology"
            onClick={[Function]}
          >
            technology
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  >
    <section
      className="main-content"
    >
      <h1
        className="page-title"
      >
        general
         News
      </h1>
      <section
        className="news-grid news-grid--loaded"
      >
        <p>
          We are having trouble fetching your news.
        </p>
      </section>
    </section>
  </main>
</div>
`;
